pkgs/development/compilers/llvm/17/libcxx/default.nix

```diff
@@ -54,6 +54,14 @@ stdenv.mkDerivation rec {
       hash = "sha256-LNoPg1KCoP8RWxU/AzHR52f4Dww24I9BGQJedMhFxyQ=";
       relative = "libcxx";
     })
+  ] ++ lib.optionals (stdenv.isDarwin && stdenv.isx86_64) [
+    # https://github.com/llvm/llvm-project/issues/64226
+    (fetchpatch {
+      name = "0042-mbstate_t-not-defined.patch";
+      url = "https://github.com/macports/macports-ports/raw/acd8acb171f1658596ed1cf25da48d5b932e2d19/lang/llvm-17/files/0042-mbstate_t-not-defined.patch";
```

Comment:
macports is only patching older systems. https://github.com/macports/macports-ports/commit/acd8acb171f1658596ed1cf25da48d5b932e2d19  can we modify the patch to only add the extra include for when the build is know to fail? so when a newer framework is used it doesn't get the extra include?

the patch applies to llvmPackages_git too. been using both to test stuff -- thanks!